rafx-base/src/atomic_once_cell.rs

```diff
@@ -0,0 +1,434 @@
+use core::ptr;
+use std::cell::UnsafeCell;
+use std::mem::MaybeUninit;
+use std::sync::atomic::{AtomicUsize, Ordering};
+
+const SET_ACQUIRE_FLAG: usize = 1 << 1;
+const SET_RELEASE_FLAG: usize = 1 << 0;
+const IS_INIT_BITMASK: usize = SET_ACQUIRE_FLAG | SET_RELEASE_FLAG;
+
+pub struct AtomicOnceCell<T> {
```

Comment:
I imagine you already have this in your mental TODOs but we should summarize the purpose of these containers and what problem they are meant to solve in the higher level code

Reply:
Is the documentation on `AtomicOnceCellArray` and `AtomicOnceCellStack` acceptable? If so, I can write a similar doc comment on `AtomicOnceCell`. If you want something more specific / different, can you give an example?

Reply:
Something like what's on those two structs is fine. What I had in mind was something really short like "AtomicOnceCell is a value that can only be set once, using an atomic value to synchronize across threads rather than a mutex." Completely up to you, happy either way.